File: Examples/VisualBasic/ModifyingAndConvertingImages/BlurAnImage.vb
Imports Aspose.Imaging.ImageFilters.FilterOptions

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages
    Class BlurAnImage
        Public Shared Sub Run()
            ' To get proper output please apply a valid Aspose.Imaging License. You can purchase full license or get 30 day temporary license from http:// Www.aspose.com/purchase/default.aspx.")
            ' ExStart:BlurAnImage
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_ModifyingAndConvertingImages()

            ' Load the image
            Using image__1 As Image = Image.Load(dataDir & Convert.ToString("asposelogo.gif"))
                ' Convert the image into RasterImage.
                Dim rasterImage As RasterImage = DirectCast(image__1, RasterImage)

                ' Pass Bounds[rectangle] of image and GaussianBlurFilterOptions instance to Filter method.
                rasterImage.Filter(rasterImage.Bounds, New GaussianBlurFilterOptions(5, 5))

                ' Save the results to output path.
                rasterImage.Save(dataDir + "BlurAnImage_out.gif")
            End Using
            ' ExEnd:BlurAnImage
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/BlurAnImage.cs
using Aspose.Imaging;
using Aspose.Imaging.ImageFilters.FilterOptions;
using System;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API reference 
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, 
install it, and then add its reference to this project. For any issues, questions, or suggestions, 
please feel free to contact us using https://forum.aspose.com/.
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages
{
    class BlurAnImage
    {
        public static void Run()
        {
            Console.WriteLine("Running example BlurAnImage");
            // To get proper output, please apply a valid Aspose.Imaging license. You can purchase a full license or obtain a 30‑day temporary license from https://www.aspose.com/purchase/default.aspx.
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ModifyingAndConvertingImages();

            // Load the image.
            using (Image image = Image.Load(dataDir + "asposelogo.gif"))
            {
                // Convert the image to a RasterImage, pass the image bounds and a GaussianBlurFilterOptions instance to the Filter method, and save the result.
                RasterImage rasterImage = (RasterImage)image;
                rasterImage.Filter(rasterImage.Bounds, new GaussianBlurFilterOptions(5, 5));
                rasterImage.Save(dataDir + "BlurAnImage_out.gif");
            }

            Console.WriteLine("Finished example BlurAnImage");
        }
    }
}